tests/libinfra/certificates.go

```diff
@@ -40,3 +49,40 @@ func ContainsCrt(bundle []byte, containedCrt []byte) bool {
 	}
 	return attached
 }
+
+func GetBundleFromConfigMap(ctx context.Context, configMapName string) ([]byte, []*x509.Certificate) {
+	virtClient := kubevirt.Client()
+	configMap, err := virtClient.CoreV1().ConfigMaps(flags.KubeVirtInstallNamespace).Get(ctx, configMapName, v1.GetOptions{})
+	gomega.Expect(err).ToNot(gomega.HaveOccurred())
+	if rawBundle, ok := configMap.Data[components.CABundleKey]; ok {
+		crts, err := cert.ParseCertsPEM([]byte(rawBundle))
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		return []byte(rawBundle), crts
+	}
+	return nil, nil
+}
+
+// EnsurePodsCertIsSynced waits until new certificates are rolled out to all pods
+// that are matching the specified labelselector.
+// Once all certificates are in sync, the final secret is returned
+func EnsurePodsCertIsSynced(labelSelector string, namespace string, port string) []byte {
+	var certs [][]byte
+	gomega.EventuallyWithOffset(1, func() bool {
+		var err error
+		certs, err = libpod.GetCertsForPods(labelSelector, namespace, port)
+		gomega.Expect(err).ToNot(gomega.HaveOccurred())
+		if len(certs) == 0 {
+			return true
+		}
+		for _, crt := range certs {
+			if !reflect.DeepEqual(certs[0], crt) {
+				return false
+			}
+		}
```

Comment:
in the first loop, we're comparing `certs[0]` to itself. since we know the slice is not empty (line 74), we can safely do:
```suggestion
		for _, crt := range certs[1:] {
			if !reflect.DeepEqual(certs[0], crt) {
				return false
			}
		}
```

Reply:
It's true logically but I think it's better to keep it general, we're not concerned about performance here, but durability of the code.

Reply:
I think that the slice syntax makes it too easy tp not doing that

Reply:
Changed.